### ISSUE TITLE
Feature Request - customizable columns in the agent list

### ISSUE THREAD

**Issue by rtwright68** [wh1te909/tacticalrmm#308]
One particular my guys mention is the LAN IP address.  That would be very helpful to be able to display that without clicking on the agent?

**Comment by dinger1986** [comment]
So you mean a column where you can select from a drop down list different options one of them being the IP address of the machine?

**Comment by rtwright68** [comment]
Sure, however it could be best implemented.

**Comment by djsvi** [comment]
Agree, customizable columns would be very helpful.

Not that you are trying to copy the major players, but for example Kaseya, N-able and Labtech all implement 'views' where you can choose which database fields you wish to see, whether these fields should have any particular filters applied to them, etc. You can save these views and easily switch back and forth between them, allowing you to quickly bring up a (potentially filtered) list of the machines you want to work with, along with the relevant fields from the database. It's super powerful.

Example view: Workstations missing Windows updates
Example columns: Machine Name, OS, Number of missing updates, Last time Updates Checked, Last time Updates Installed.
Example filter linked to the view: OS type = Workstation, Number of missing updates > 0

**Comment by sadnub** [comment]
#219 Merging with this request

**Comment by plao** [comment]
I TOTALLY agree! It would be really helpful to see if a client are connected from home (in VPN), filtering the public IPs of the 2 offices...

**Comment by jdustinlewis** [comment]
This would be really handy, especially for adding custom fields to the agent table!

**Comment by scotbf** [comment]
We are also very interested in displaying custom fields on the agent table. In general, of course, to be able to edit the agent table.

**Comment by silversword411** [comment]
Step 1 is done, customizable columns library: https://github.com/amidaware/tacticalrmm-web/pull/9

Next step is putting it into the agents table, and then adding custom fields into selectable list

**Comment by olefm-dev** [comment]
Would love to see this feature come to life 👍

**Comment by muzzol** [comment]
Is there any work on this?

I really need a custom column. Clicking on 300 machines to get a value is terrible for my carpal tunnel and for my mental sanity.

Can I try develop branch to see this feature?

**Comment by P6g9YHK6** [comment]
> Is there any work on this?
> 
> I really need a custom column. Clicking on 300 machines to get a value is terrible for my carpal tunnel and for my mental sanity.
> 
> Can I try develop branch to see this feature?

script a report to a random table
